### PR TITLE
Todo list

### PR DESCRIPTION
**Pull request by LeoKuD**
[Link to the deploy](https://leokud.github.io/todo-list/)

**Done:** 

- adding, deleting, editing tasks;
- storage tasks in localStorage;
- displaying the number of unfinished and completed tasks;
- dark theme - with save after reload;
- sorting - with save after reload.

### CODE REVIEW

**Comment by Dev1LtJ** [comment]
@LeoKuD, hello. Normalized is linked by default on HTML. You can delete it. Also, please, upload the original HTML file main branch.

File: js/index.js [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+});
+
+function deleteTask(itemID) {
+  tasks = tasks.filter((item) => item.id !== itemID);
+  updateLocalStorage(localStorageValues.tasks, tasks);
+  fillHTMLList();
+}
+function editTask(index) {
+  showModal();
+  myForm.removeEventListener('submit', taskSubmit);
```
Comment (Dev1LtJ) [inline js/index.js:201]: We have discussed, that you don't need to always remove and add event listeners here. This logic is vulnerable to bugs. Try to change it.

File: js/index.js [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+  titleHeaderForm.innerText = `Edit task`;
+
+  myForm.elements.title.value = tasks[index].title;
+  myForm.elements.text.value = tasks[index].text;
+  myForm.elements.priority.forEach((item) => {
+    item.id === tasks[index].priority
+      ? (item.checked = true)
+      : (item.checked = false);
+  });
+  listenerSubmitEditTask = submitEditTask.bind(null, index);
```
Comment (Dev1LtJ) [inline js/index.js:211]: try to avoid this constructions.

File: js/index.js [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+  showModal();
+  myForm.removeEventListener('submit', taskSubmit);
+  titleHeaderForm.innerText = `Edit task`;
+
+  myForm.elements.title.value = tasks[index].title;
+  myForm.elements.text.value = tasks[index].text;
+  myForm.elements.priority.forEach((item) => {
+    item.id === tasks[index].priority
+      ? (item.checked = true)
+      : (item.checked = false);
```
Comment (Dev1LtJ) [inline js/index.js:209]: You can short this code.

File: js/index.js [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+  btnAddTask.click();
+  btnCloseForm.innerText = 'Cancel';
+  btnSubmitForm.innerText = 'Save';
+}
+
+exampleModal.addEventListener('click', (e) => {
+  if (
+    e.target.id === 'exampleModal' ||
+    e.target.id === 'btn-close' ||
+    e.target.id === 'close-right'
```
Comment (Dev1LtJ) [inline js/index.js:238]: It's better to create enums

File: js/index.js [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+  fillHTMLList();
+}
+
+content.addEventListener('click', (e) => {
+  if (e.target.id === 'btn-delete') {
+    deleteTask(e.target.getAttribute('data-index'));
+  } else if (e.target.id === 'btn-completed') {
+    completeTask(e.target.getAttribute('data-index'));
+  } else if (e.target.id === 'btn-edit') {
+    editTask(e.target.getAttribute('data-index'));
```
Comment (Dev1LtJ) [inline js/index.js:170]: It's better to create enums

File: js/index.js [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+}
+
+const updateLocalStorage = (key, value) => {
+  localStorage.setItem(key, JSON.stringify(value));
+};
+
+btnAddTask.addEventListener('click', () => {
+  btnCloseForm.innerText = 'Close';
+  btnSubmitForm.innerText = 'Add Task';
+  titleHeaderForm.innerText = `Add task`;
```
[the remark this answers is not in the document]
Comment (Dev1LtJ) [reply js/index.js:143]: It's better to create enums

File: js/index.js [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+}
+
+exampleModal.addEventListener('click', (e) => {
+  if (
+    e.target.id === 'exampleModal' ||
+    e.target.id === 'btn-close' ||
+    e.target.id === 'close-right'
+  ) {
+    myForm.removeEventListener('submit', listenerSubmitEditTask);
+    myForm.removeEventListener('submit', taskSubmit);
```
Comment (Dev1LtJ) [inline js/index.js:241]: We have discussed, that you don't need to always remove and add event listeners here. This logic is vulnerable to bugs. Try to change it.

File: scss/style.scss [hunk 8]
```diff
@@ -0,0 +1,132 @@
+$bc-color-primary: #1f1d36;
+$bc-color-secondary: #3f3351;
+$bc-color-tertiary: #e9a6a6;
+$priority-color-primary-light: green;
+$priority-color-secondary-light: yellow;
+$priority-color-tertiary-light: orange;
+$bc-color-primary-light: #fff;
```
Comment (Dev1LtJ) [inline scss/style.scss:7]: $b**g**-color-... if you use it as background-color

File: scss/style.scss [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+  border: 1px solid #bbc1e1;
+  border-radius: 30px;
+  outline: none;
+  cursor: pointer;
+  appearance: none;
+  transition: $transition;
+}
+
+.toggle-button::after {
+  transition: $transition / 1;
```
Comment (Dev1LtJ) [inline scss/style.scss:110]: 0.5 / 1 = 0.5. Why do you do it?

**Review by Dev1LtJ — CHANGES_REQUESTED** [review]
Hello! Good work. I left some comments. You didn't resolve all that we previously discussed. Fix it, please.

File: .gitignore [hunk 10]
```diff
@@ -0,0 +1,2 @@
+/node_modules
+package-lock.json
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply .gitignore:2]: fix

File: js/index.js [hunk 11]
```diff
@@ -0,0 +1,258 @@
...
+const dark = 'dark';
+const light = 'light';
+const up = 'up';
+const down = 'down';
+let tasks;
+let listenerSubmitEditTask = null;
+let defaultSort;
+let defaultTheme;
+
+const localStorageValues =  {
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:26]:
```
const LOCALSTORAGEKEYS =  {
  tasks: 'tasks',
  sorting: 'sorting',
  theme: 'theme'
}
```

File: js/index.js [hunk 12]
```diff
@@ -0,0 +1,258 @@
...
+
+const body = document.body;
+const dark = 'dark';
+const light = 'light';
+const up = 'up';
+const down = 'down';
+let tasks;
+let listenerSubmitEditTask = null;
+let defaultSort;
+let defaultTheme;
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:24]:
```
let selectedSort;
let selectedTheme;
```

File: js/index.js [hunk 13]
```diff
@@ -0,0 +1,258 @@
...
+const titleHeaderForm = document.querySelector('.modal-title');
+const exampleModal = document.getElementById('exampleModal');
+const btnSortUp = document.getElementById('sort-up');
+const btnSortDown = document.getElementById('sort-down');
+
+const body = document.body;
+const dark = 'dark';
+const light = 'light';
+const up = 'up';
+const down = 'down';
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:20]:
```
const KEYSVARIABLES = {
  dark: 'dark',
  light: 'light',
  up: 'up',
  down: 'down'
}
```

File: js/index.js [hunk 14]
```diff
@@ -0,0 +1,258 @@
...
+
+const localStorageValues =  {
+  tasks: 'tasks',
+  sorting: 'sorting',
+  theme: 'theme'
+}
+
+!localStorage[localStorageValues.tasks]
+  ? (tasks = [])
+  : (tasks = JSON.parse(localStorage.getItem(localStorageValues.tasks)));
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:34]:
```
tasks = JSON.parse(localStorage.getItem(LOCALSTORAGEKEYS.tasks)) || [];
selectedSort = JSON.parse(localStorage.getItem(LOCALSTORAGEKEYS.sorting)) || LOCALSTORAGEKEYS.up;
```

File: js/index.js [hunk 15]
```diff
@@ -0,0 +1,258 @@
...
+
+!localStorage[localStorageValues.sorting]
+  ? (defaultSort = up)
+  : (defaultSort = JSON.parse(localStorage.getItem(localStorageValues.sorting)));
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:38]: `selectedSort = JSON.parse(localStorage.getItem(LOCALSTORAGEKEYS.sorting)) || LOCALSTORAGEKEYS.up;`

File: js/index.js [hunk 16]
```diff
@@ -0,0 +1,258 @@
...
+
+const makeId = () => {
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:40]:
```
const generateId = () => {
  let ID = '';
```

File: js/index.js [hunk 17]
```diff
@@ -0,0 +1,258 @@
...
+  let ID = '';
+  let characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
+  for (let i = 0; i < 12; i++) {
+    ID += characters.charAt(Math.floor(Math.random() * 36));
+  }
+  return ID;
+};
+
+function Task([title, text, priority]) {
+  this.id = makeId();
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:50]:
```
function Task([title, text, priority]) {
  this.id = generateId();
```

File: js/index.js [hunk 18]
```diff
@@ -0,0 +1,258 @@
...
+  ? (tasks = [])
+  : (tasks = JSON.parse(localStorage.getItem(localStorageValues.tasks)));
+
+!localStorage[localStorageValues.sorting]
+  ? (defaultSort = up)
+  : (defaultSort = JSON.parse(localStorage.getItem(localStorageValues.sorting)));
+
+const makeId = () => {
+  let ID = '';
+  let characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:42]: `  const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
`

File: js/index.js [hunk 19]
```diff
@@ -0,0 +1,258 @@
...
+        !item.isDone ? listCurrentTasks.insertAdjacentHTML('beforeend', taskTemplate(item, index))
+          : listCompletedTasks.insertAdjacentHTML('beforeend', taskTemplate(item, index));
+      });
+    } else {
+      tasks.forEach((item, index) => {
+        !item.isDone ? listCurrentTasks.insertAdjacentHTML('afterbegin', taskTemplate(item, index))
+          : listCompletedTasks.insertAdjacentHTML('afterbegin',taskTemplate(item, index));
+      });
+    }
+  }
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:73]:
```
  if (tasks.length) {
    tasks.forEach((item, index) => {
      const taskItem = taskTemplate(item, index);
      selectedSort === KEYSVARIABLES.up ?
        (!item.isDone
          ? listCurrentTasks.insertAdjacentHTML('beforeend', taskItem)
          : listCompletedTasks.insertAdjacentHTML('beforeend', taskItem))
      :
        (!item.isDone
          ? listCurrentTasks.insertAdjacentHTML('afterbegin', taskItem)
          : listCompletedTasks.insertAdjacentHTML('afterbegin', taskItem))
      
    });
  }
```

File: js/index.js [hunk 20]
```diff
@@ -0,0 +1,258 @@
...
+  const countCompletedTasks = tasks.filter(
+    (item) => item.isDone === true
+  ).length;
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:76]: `  const countCompletedTasks = tasks.filter((item) => item.isDone).length;
`

File: js/index.js [hunk 21]
```diff
@@ -0,0 +1,258 @@
...
+  titleTodoBlock.textContent = `ToDo (${tasks.length - countCompletedTasks})`;
+  titleCompletedBlock.textContent = `Completed (${countCompletedTasks})`;
+}
+
+function getDate(stamp) {
+  const date = new Date(stamp);
+  const month = date.getMonth() < 9 ? `0${date.getMonth()+1}` : date.getMonth()+1;
+  const day = date.getDate() < 10 ? `0${date.getDate()}` : date.getDate();
+  return ` ${date.getHours()}:${date.getMinutes()} ${day}.${month}.${date.getFullYear()}`;
+}
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:86]:
```
function checkDate(date) {
  if (date < 10) {
    return `0${date}`
  }
  return date
}

function getDate(stamp) {
  const date = new Date(stamp);
  const month =
    date.getMonth() < 9 ? `0${date.getMonth() + 1}` : date.getMonth() + 1;
  const day = checkDate(date.getDate());
  const hours = checkDate(date.getHours());
  const minutes = checkDate(date.getMinutes());
  const fullYear = date.getFullYear();
  return ` ${hours}:${minutes} ${day}.${month}.${fullYear}`;
}
```

File: js/index.js [hunk 22]
```diff
@@ -0,0 +1,258 @@
...
+                  </button>
+                </div>
+              </div>
+            </li>
+    `;
+}
+
+const updateLocalStorage = (key, value) => {
+  localStorage.setItem(key, JSON.stringify(value));
+};
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:138]: fix

File: js/index.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:143]:
```

btnAddTask.addEventListener('click', () => {
  btnCloseForm.innerText = KEYSVARIABLES.close;
  btnSubmitForm.innerText = KEYSVARIABLES.addTaks;
  titleHeaderForm.innerText = KEYSVARIABLES.addTaks;
  myForm.addEventListener('submit', taskSubmit);
});
```

File: js/index.js [hunk 23]
```diff
@@ -0,0 +1,258 @@
...
+  body.className === light
+    ? ((defaultTheme = dark),
+      body.classList.remove(light),
+      body.classList.add(dark),
+      updateLocalStorage(localStorageValues.theme, defaultTheme))
+    : ((defaultTheme = light),
+      body.classList.remove(dark),
+      body.classList.add(light),
+      updateLocalStorage(localStorageValues.theme, defaultTheme));
+});
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:184]:
```
toggleTheme.addEventListener('click', () => {
  body.classList.toggle(selectedTheme);
  selectedTheme === KEYSVARIABLES.light
    ? (selectedTheme = KEYSVARIABLES.dark)
    : (selectedTheme = KEYSVARIABLES.light);
  body.classList.toggle(selectedTheme);
  updateLocalStorage(LOCALSTORAGEKEYS.theme, selectedTheme);
});
```

File: js/index.js [hunk 24]
```diff
@@ -0,0 +1,258 @@
...
+
+document.addEventListener('DOMContentLoaded', () => {
+  !localStorage[localStorageValues.theme]
+    ? (defaultTheme = light)
+    : (defaultTheme = JSON.parse(localStorage.getItem(localStorageValues.theme)));
+
+  body.classList.add(defaultTheme);
+});
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:192]:
```
document.addEventListener('DOMContentLoaded', () => {
  selectedTheme =
    JSON.parse(localStorage.getItem(LOCALSTORAGEKEYS.theme)) ||
    KEYSVARIABLES.light;
  body.classList.add(selectedTheme);
});

```

File: css/style.css [hunk 25]
```diff
@@ -0,0 +1,115 @@
...
+}
+.light .Low {
+  background-color: green;
+}
+.light .Medium {
+  background-color: yellow;
+}
+.light .High {
+  background-color: orange;
+}
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply css/style.css:76]:
```
.light {
  transition: $transition;
  .Low {
    background-color: $priority-color-primary-light;
  }

  .Medium {
    background-color: $priority-color-secondary-light;
  }

  .High {
    background-color: $priority-color-tertiary-light;
  }
}
```

File: scss/style.scss (hunk 9 above)
Comment (LeoKuD) [reply scss/style.scss:110]: fix

File: js/index.js [hunk 26]
```diff
@@ -0,0 +1,258 @@
...
+}
+
+function getDate(stamp) {
+  const date = new Date(stamp);
+  const month = date.getMonth() < 9 ? `0${date.getMonth()+1}` : date.getMonth()+1;
+  const day = date.getDate() < 10 ? `0${date.getDate()}` : date.getDate();
+  return ` ${date.getHours()}:${date.getMinutes()} ${day}.${month}.${date.getFullYear()}`;
+}
+
+function taskTemplate(item, index) {
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:88]:
```
    tasks.forEach((item, index) => {
      const taskItem = taskTemplate(item, index);
      selectedSort === KEYSVARIABLES.up
        ? !item.isDone
          ? listCurrentTasks.append(taskItem)
          : listCompletedTasks.append(taskItem)
        : !item.isDone
        ? listCurrentTasks.prepend(taskItem)
        : listCompletedTasks.prepend(taskItem);
    });
```

```
  const taskNode = document.createElement('li');
  taskNode.className = `list-group-item d-flex w-100 mb-2 ${item.priority}`;
  taskNode.innerHTML = `<div class="w-100 mr-2">
```

File: js/index.js (hunk 1 above)
Comment (LeoKuD) [reply js/index.js:201]:
```
myForm.addEventListener('submit', (e) => {
  e.preventDefault()
  console.log(e.submitter.id);
    if (e.submitter.id === btnSubmitForm.id) {
      taskSubmit();
    } else if (e.submitter.id === btnSubmitEditForm.id){
      listenerSubmitEditTask()
    }
})
```

Now only one event listener submit

File: js/index.js (hunk 2 above)
Comment (LeoKuD) [reply js/index.js:211]:
```
myForm.addEventListener('submit', (e) => {
  e.preventDefault();
  if (e.submitter.id === btnSubmitForm.id) {
    taskSubmit();
  } else if (e.submitter.id === btnSubmitEditForm.id) {
    submitEditTask(e.submitter.getAttribute('data-index'));
  }
});
```

File: js/index.js (hunk 4 above)
Comment (LeoKuD) [reply js/index.js:238]:
```
 if (
    e.target.id === KEYSVARIABLES.exampleModal ||
    e.target.id === KEYSVARIABLES.btnClose ||
    e.target.id === KEYSVARIABLES.closeRight
  ) {
```

File: js/index.js (hunk 7 above)
Comment (LeoKuD) [reply js/index.js:241]: Fixed

File: scss/style.scss (hunk 8 above)
Comment (LeoKuD) [reply scss/style.scss:7]:
```
$bg-color-primary: #1f1d36;
$bg-color-secondary: #3f3351;
$bg-color-tertiary: #e9a6a6;
$priority-color-primary-light: green;
$priority-color-secondary-light: yellow;
$priority-color-tertiary-light: orange;
$bg-color-primary-light: #fff;
$transition: 0.5s;
```
Fixed

File: css/style.css [hunk 27]
```diff
@@ -0,0 +1,112 @@
...
```
Comment (Dev1LtJ) [inline css/style.css:73]: You didn't change.

File: css/style.css [hunk 28]
```diff
@@ -0,0 +1,112 @@
...
+  content: "";
+  display: inline-block;
+  position: absolute;
+  left: 3px;
+  top: 1.5px;
+  width: 20px;
+  height: 20px;
+  background-color: blue;
+  border-radius: 50%;
+  transform: translateX(0);
```
Comment (Dev1LtJ) [inline css/style.css:102]: What does it do?

File: index.html [hunk 29]
```diff
@@ -1,142 +1,273 @@
...
+      crossorigin="anonymous"
+    ></script>
     <style>
-    .wrapper {
+      .wrapper {
         min-height: calc(100vh - 15px);
         margin: 15px auto 0;
-    }
+      }
     </style>
```
Comment (Dev1LtJ) [inline index.html:25]: Try to not use this type of styles.

File: js/index.js [hunk 30]
```diff
@@ -0,0 +1,282 @@
...
+
+function fillHTMLList() {
+  listCurrentTasks.textContent = '';
+  listCompletedTasks.textContent = '';
+  if (tasks.length) {
+    tasks.forEach((item, index) => {
+      const taskItem = taskTemplate(item, index);
+      selectedSort === KEYSVARIABLES.up
+        ? !item.isDone
+          ? listCurrentTasks.append(taskItem)
```
Comment (Dev1LtJ) [inline js/index.js:102]: Do not use double `?` It's harder to read.

File: js/index.js (hunk 26 above)
[the remark this answers is not in the document]
Comment (Dev1LtJ) [reply js/index.js:132]: I meant function name `createTaskNode(item, index)`, but your code is good too.

File: js/index.js [hunk 31]
```diff
@@ -0,0 +1,282 @@
...
+  myForm['btn-close'].click();
+}
+
+function showModal(index) {
+  btnAddTask.click();
+  btnCloseForm.innerText = KEYSVARIABLES.cancel;
+  btnSubmitEditForm.innerText = KEYSVARIABLES.save;
+  btnSubmitEditForm.style.display = 'inline-block';
+  btnSubmitEditForm.setAttribute(KEYSVARIABLES.dataIndex, index);
+  btnSubmitForm.style.display = 'none';
```
Comment (Dev1LtJ) [inline js/index.js:254]: Try to not styling with JS if it is not really needed.

File: js/index.js [hunk 32]
```diff
@@ -0,0 +1,282 @@
...
+
+exampleModal.addEventListener('click', (e) => {
+  if (
+    e.target.id === KEYSVARIABLES.exampleModal ||
+    e.target.id === KEYSVARIABLES.btnClose ||
+    e.target.id === KEYSVARIABLES.closeRight
+  ) {
+    myForm.reset();
+    btnSubmitForm.style.display = 'inline-block';
+    btnSubmitEditForm.style.display = 'none';
```
Comment (Dev1LtJ) [inline js/index.js:266]: Try to not styling with JS if it is not really needed.

**Review by Dev1LtJ — CHANGES_REQUESTED** [review]
Hello! Almost done. Fix, what you didn't fix before (theme and rename function) and some comments, what I left for your with this review.

File: js/index.js (hunk 19 above)
[the remark this answers is not in the document]
Comment (Dev1LtJ) [reply js/index.js]: Answered on new review.

File: js/index.js (hunk 23 above)
[the remark this answers is not in the document]
Comment (Dev1LtJ) [reply js/index.js]: Don't fixed for now.

File: css/style.css (hunk 25 above)
[the remark this answers is not in the document]
Comment (Dev1LtJ) [reply css/style.css]: Don't fixed for now.

File: js/index.js (hunk 3 above)
Comment (LeoKuD) [reply js/index.js:233]: `  myForm.elements[tasks[index].priority].checked = true;`

File: js/index.js (hunk 30 above)
Comment (LeoKuD) [reply js/index.js:102]:
```
      if (selectedSort === KEYSVARIABLES.up) {
        !item.isDone
          ? listCurrentTasks.append(taskItem)
          : listCompletedTasks.append(taskItem);
      } else {
        !item.isDone
          ? listCurrentTasks.prepend(taskItem)
          : listCompletedTasks.prepend(taskItem);
      }
```

File: js/index.js (hunk 19 above)
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js]: Fiexed on new review.

File: js/index.js (hunk 23 above)
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js]:
```
toggleTheme.addEventListener('click', () => {
  body.classList.toggle(KEYSVARIABLES.dark);
  body.classList.contains(KEYSVARIABLES.dark)
    ? updateLocalStorage(LOCALSTORAGEKEYS.theme, KEYSVARIABLES.dark)
    : localStorage.removeItem(LOCALSTORAGEKEYS.theme);
});
```

[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js]:
```
function setTheme() {
  body.classList.contains(KEYSVARIABLES.dark)
    ? updateLocalStorage(LOCALSTORAGEKEYS.theme, KEYSVARIABLES.dark)
    : localStorage.removeItem(LOCALSTORAGEKEYS.theme);
}

toggleTheme.addEventListener('click', () => {
  body.classList.toggle(KEYSVARIABLES.dark);
  setTheme();
});
```

File: css/style.css (hunk 25 above)
[the remark this answers is not in the document]
Comment (LeoKuD) [reply css/style.css]:
```
  .Low {
    background-color: $priority-color-primary-light;
  }

  .Medium {
    background-color: $priority-color-secondary-light;
  }

  .High {
    background-color: $priority-color-tertiary-light;
  }
```

[the remark this answers is not in the document]
Comment (LeoKuD) [reply css/style.css]: Now whitout light

File: index.html (hunk 29 above)
Comment (LeoKuD) [reply index.html:25]: Ok, but these styles were on the default pages

File: css/style.css (hunk 27 above)
Comment (LeoKuD) [reply css/style.css:73]:
```
  .Low {
    background-color: $priority-color-primary-light;
  }

  .Medium {
    background-color: $priority-color-secondary-light;
  }

  .High {
    background-color: $priority-color-tertiary-light;
  }
```

Comment (LeoKuD) [reply css/style.css:73]: Now without light

File: css/style.css (hunk 28 above)
Comment (LeoKuD) [reply css/style.css:102]: Sorry, this is not needed here. Copy-Past!

File: js/index.js (hunk 32 above)
Comment (LeoKuD) [reply js/index.js:266]:
```
  btnSubmitForm.classList.add(KEYSVARIABLES.none);
  btnSubmitForm.classList.remove(KEYSVARIABLES.inlineBlock);
```

File: index.html (hunk 29 above)
Comment (maximsan) [reply index.html:25]: in any case it's better place styles in file

**Review by Dev1LtJ — APPROVED** [review]
Approved. Wait for @maximsan's review.
P.S. You have not renamed function for now. Rename `taskTemplate` to `createTaskNode` during @maximsan's review.

**Review by maximsan — CHANGES_REQUESTED** [review]
Hello! I left small comments, please look them through. Ping me when it's ready and I'll approve it. It looks promising 🔥

File: package.json [hunk 33]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "todo",
+  "version": "1.0.0",
+  "description": "todo list",
+  "main": "gulpfile.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply package.json:7]: Fix

File: scss/style.scss [hunk 34]
```diff
@@ -0,0 +1,132 @@
...
+  .Medium {
+    background-color: $bg-color-secondary !important;
+  }
+
+  .High {
+    background-color: darken(
+      $color: $bg-color-secondary,
+      $amount: 15%
+    ) !important;
+  }
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply scss/style.scss:34]: Corrected.

File: scss/style.scss [hunk 35]
```diff
@@ -0,0 +1,132 @@
...
+    background-color: $priority-color-primary-light;
+  }
+
+  .Medium {
+    background-color: $priority-color-secondary-light;
+  }
+
+  .High {
+    background-color: $priority-color-tertiary-light;
+  }
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply scss/style.scss:85]:
```
  .high {
    background-color: $priority-color-tertiary-light;
  }
```

File: js/index.js [hunk 36]
```diff
@@ -0,0 +1,277 @@
...
+  dataIndex: 'data-index',
+  btnCompleted: 'btn-completed',
+  btnEdit: 'btn-edit',
+  save: 'Save',
+  exampleModal: 'exampleModal',
+  btnClose: 'btn-close',
+  closeRight: 'close-right',
+  inlineBlock: 'inline-block',
+  none: 'none',
+};
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:48]: const VARIABLESTEXTKEYS = {
  close: 'Close',
  cancel: 'Cancel',
  addTaks: 'Add task',
  editTaks: 'Edit task',
  dataIndex: 'data-index',
  inlineBlock: 'inline-block',
  none: 'none',
  save: 'Save',
};
const ELEMENTSKEYS = {
  btnDelete: 'btn-delete',
  btnCompleted: 'btn-completed',
  btnEdit: 'btn-edit',
  exampleModal: 'exampleModal',
  btnClose: 'btn-close',
};

[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:48]:
```
const LOCALSTORAGEKEYS = {
  tasks: 'tasks',
  sorting: 'sorting',
  theme: 'theme',
  dark: 'dark',
  up: 'up',
  down: 'down',
};

const VARIABLESTEXTKEYS = {
  close: 'Close',
  cancel: 'Cancel',
  addTaks: 'Add task',
  editTaks: 'Edit task',
  dataIndex: 'data-index',
  inlineBlock: 'inline-block',
  none: 'none',
  save: 'Save',
};
const ELEMENTSKEYS = {
  btnDelete: 'btn-delete',
  btnCompleted: 'btn-completed',
  btnEdit: 'btn-edit',
  exampleModal: 'exampleModal',
  btnClose: 'btn-close',
};
```

[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:48]:
```
const LOCALSTORAGEKEYS = {
  tasks: 'tasks',
  sorting: 'sorting',
  theme: 'theme',
  dark: 'dark',
  up: 'up',
  down: 'down',
};

const ELEMETSTEXTKEYS = {
  close: 'Close',
  cancel: 'Cancel',
  addTaks: 'Add task',
  editTaks: 'Edit task',
  save: 'Save',
};

const STYLESKEYS = {
  dataIndex: 'data-index',
  inlineBlock: 'inline-block',
  none: 'none',
};

const ELEMENTSKEYS = {
  btnDelete: 'btn-delete',
  btnCompleted: 'btn-completed',
  btnEdit: 'btn-edit',
  exampleModal: 'exampleModal',
  btnClose: 'btn-close',
};
```

File: js/index.js [hunk 37]
```diff
@@ -0,0 +1,277 @@
+const myForm = document.forms[0];
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:1]:
```suggestion
const myForm = document.getElementById('my-form');
```

[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:1]:
```suggestion
const myForm = document.querySelector('.my-form');
```

File: js/index.js [hunk 38]
```diff
@@ -0,0 +1,277 @@
...
+const listCurrentTasks = document.getElementById('currentTasks');
+const listCompletedTasks = document.getElementById('completedTasks');
+const titleTodoBlock = document.getElementById('title-todo-block');
+const titleCompletedBlock = document.getElementById('title-completed-block');
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:5]: Ok, thanks

File: js/index.js [hunk 39]
```diff
@@ -0,0 +1,277 @@
...
+document.addEventListener('DOMContentLoaded', () => {
+  if (
+    JSON.parse(localStorage.getItem(LOCALSTORAGEKEYS.theme)) ===
+    KEYSVARIABLES.dark
+  ) {
+    body.classList.add(KEYSVARIABLES.dark);
+  }
+});
+
+btnSubmitEditForm.id = 'btnSubmitEditForm';
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:63]:
```
const btnSubmitEditForm = btnSubmitForm.cloneNode(true);
btnSubmitEditForm.id = 'btnSubmitEditForm';
```

File: js/index.js [hunk 40]
```diff
@@ -0,0 +1,277 @@
...
+});
+
+tasks = JSON.parse(localStorage.getItem(LOCALSTORAGEKEYS.tasks)) || [];
+selectedSort =
+  JSON.parse(localStorage.getItem(LOCALSTORAGEKEYS.sorting)) ||
+  LOCALSTORAGEKEYS.up;
+
+const generateId = () => {
+  let ID = '';
+  const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:81]: Fixed

File: js/index.js [hunk 41]
```diff
@@ -0,0 +1,277 @@
...
+  const countCompletedTasks = tasks.filter((item) => item.isDone).length;
+  titleTodoBlock.textContent = `ToDo (${tasks.length - countCompletedTasks})`;
+  titleCompletedBlock.textContent = `Completed (${countCompletedTasks})`;
+}
+
+function checkDate(date) {
+  if (date < 10) {
+    return `0${date}`;
+  }
+  return date;
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:123]:
```suggestion
  function checkDate(date) {
  return date < 10 ? `0${date}`: date;
}
```

[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:123]:
```suggestion
  return date;
```
```suggestion
 function checkDate(date) {
  return date < 10 ? `0${date}`: date;
}
```

[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:123]:
```suggestion
  function checkDate(date) {
  return date < 10 ? `0${date}`: date;
}
```

File: js/index.js [hunk 42]
```diff
@@ -0,0 +1,277 @@
...
+  LOCALSTORAGEKEYS.up;
+
+const generateId = () => {
+  let ID = '';
+  const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
+  for (let i = 0; i < 12; i++) {
+    ID += characters.charAt(Math.floor(Math.random() * 36));
+  }
+  return ID;
+};
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:86]:
```
const helpers = {
  generateId: function generateId() {
    let ID = '';
    for (let i = 0; i < 12; i++) {
      ID += characters.charAt(Math.floor(Math.random() * 36));
    }
    return ID;
  },
  checkDate: function checkDate(date) {
    return date < 10 ? `0${date}` : date;
  },
  getDate: function getDate(stamp) {
    const date = new Date(stamp);
    const month =
      date.getMonth() < 9 ? `0${date.getMonth() + 1}` : date.getMonth() + 1;
    const day = this.checkDate(date.getDate());
    const hours = this.checkDate(date.getHours());
    const minutes = this.checkDate(date.getMinutes());
    const fullYear = date.getFullYear();
    return ` ${hours}:${minutes} ${day}.${month}.${fullYear}`;
  },
  setTheme: function setTheme() {
    body.classList.contains(LOCALSTORAGEKEYS.dark)
      ? updateLocalStorage(LOCALSTORAGEKEYS.theme, LOCALSTORAGEKEYS.dark)
      : localStorage.removeItem(LOCALSTORAGEKEYS.theme);
  },
};
```

File: js/index.js [hunk 43]
```diff
@@ -0,0 +1,277 @@
...
+
+function Task([title, text, priority]) {
```
[the remark this answers is not in the document]
Comment (LeoKuD) [reply js/index.js:88]:
```suggestion
function Task(title, text, priority) {

```

File: js/index.js (hunk 36 above)
[the remark this answers is not in the document]
Comment (maximsan) [reply js/index.js:46]: Please add delimiters. Is it actually easy for you to read it ? Commonly we use PascalCase or
Snake Case (All Caps) - USER_LOGIN_COUNT for constants like this. It's common practice.

**Review by maximsan — APPROVED** [review]
I left 1 more comment. Everything looks good besides it. 

It's approved. Good work :)

**Pull request merged**